backend/index.ts

```diff
@@ -0,0 +1,75 @@
+import {body} from 'express-validator';
+import express, {Request, Response} from "express"
+import * as mongoose from "mongoose";
+import cors from 'cors'
+import {
+    checkAuth,
+    checkAuthorisation,
+    createPost, deletePost,
+    getAllPosts,
+    getOnePost,
+    login,
+    register, updatePost
+} from "./src/helpers/functions";
+import multer from "multer";
+import * as path from "path";
+import {v1} from "uuid";
+
+const app = express()
+app.use(express.json())
+app.use('/upload', express.static('upload')) //catch GET request to static files
+app.use(cors())
+const port = 3222
+mongoose
+    .connect('mongodb+srv://admin:admin123@cluster0.ttgwg1i.mongodb.net/blog?retryWrites=true&w=majority')
+    .then(() => console.log('DB OK'))
+    .catch((err) => console.log('DB ERROR', err));
+
+// const photoStorage = multer.diskStorage({
+//     destination:(_,_,cb)=>{
+//         cb(null,'uploads')
+//     }
+// })
+const storage = multer.diskStorage({
+    destination: (req, file, callback) => {
+        callback(null, path.resolve('./upload/'))
+    },
+    filename: (req, file, callback) => {
+        callback(null, `${v1()}-${file.originalname}`);
+    }
+});
+const upload =multer({storage})
+
+//registration
+app.post(
+    '/auth/register',
+    body('email').isEmail(),
+    body('password', "mi222n 5").isLength({min: 6}),
+    body('fullName').isLength({min: 3}),
+    register
+);
+app.post('/auth/login',
+    body('email').isEmail(),
+    body('password', "mi222n 5").isLength({min: 6}),
+    login)
+app.get('/auth/me',checkAuth,checkAuthorisation)

```

Comment:
so I think this route is completely redundant. 
The standard flow is like this:
1. User registers - he gets appended to DB, no need to give user token at this point
2. User logs in - then he is issued a token
3. On **each further request** (this is the important part) you need to check if token is valid. That is basically the point of this token. And if the token is not valid BE should refuse to answer and return an error that token is invalid. 

So there is no reason to have an endpoint just for checking the token. 
If you called it `userInfo` that would make more sense, then you could tell its an endpoint for getting user information for maybe displaying it somewhere in the application

Reply:
fixed, I keep it for displaying user information in the frontend part. Change the name to userInfo so it makes more sense